scr/main/java/bittercode/util/StoreUtil.java

```diff
@@ -0,0 +1,86 @@
+package com.bittercode.util;
+
+import java.io.PrintWriter;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+import com.bittercode.model.UserRole;
+
+/*
+ * Store UTil File To Store Commonly used methods
+ */
+public class StoreUtil {
+
+    /**
+     * Check if the User is logged in with the requested role
+     */
+    public static boolean isLoggedIn(UserRole role, HttpSession session) {
+
+        return session.getAttribute(role.toString()) != null;
+    }
+
+    /**
+     * Modify the active tab in the page menu bar
+     */
+    public static void setActiveTab(PrintWriter pw, String activeTab) {
+
+        pw.println("<script>document.getElementById(activeTab).classList.remove(\"active\");activeTab=" + activeTab
+                + "</script>");
+        pw.println("<script>document.getElementById('" + activeTab + "').classList.add(\"active\");</script>");
+
+    }
+
+    /**
+     * Add/Remove/Update Item in the cart using the session
+     */
+    public static void updateCartItems(HttpServletRequest req) {
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2438181340>#16fdcd</a><span> - 10/25/2024, 04:08 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The updateCartItems method is too long and handles multiple responsibilities, violating the Single Responsibility Principle
</div>
<div id="fix">
<b>Fix:</b> Refactor the updateCartItems method into smaller, more focused methods
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -37,49 +37,65 @@
  public static void updateCartItems(HttpServletRequest req) {
          String selectedBookId = req.getParameter("selectedBookId");
          HttpSession session = req.getSession();
 -        if (selectedBookId != null) { // add item to the cart
 -
 -            // Items will contain comma separated bookIds that needs to be added in the cart
 -            String items = (String) session.getAttribute("items");
 -            if (req.getParameter("addToCart") != null) { // add to cart
 -                if (items == null || items.length() == 0)
 -                    items = selectedBookId;
 -                else if (!items.contains(selectedBookId))
 -                    items = items + "," + selectedBookId; // if items already contains bookId, don't add it
 -
 -                // set the items in the session to be used later
 -                session.setAttribute("items", items);
 -
 -                /*
 -                 * Quantity of each item in the cart will be stored in the session as:
 -                 * Prefixed with qty_ following its bookId
 -                 * For example 2 no. of book with id 'myBook' in the cart will be
 -                 * added to the session as qty_myBook=2
 -                 */
 -                int itemQty = 0;
 -                if (session.getAttribute("qty_" + selectedBookId) != null)
 -                    itemQty = (int) session.getAttribute("qty_" + selectedBookId);
 -                itemQty += 1;
 -                session.setAttribute("qty_" + selectedBookId, itemQty);
 -            } else { // remove from the cart
 -                int itemQty = 0;
 -if (session.getAttribute("qty_" + selectedBookId) != null)
 -                    itemQty = (int) session.getAttribute("qty_" + selectedBookId);
 -                if (itemQty > 1) {
 -                    itemQty--;
 -                    session.setAttribute("qty_" + selectedBookId, itemQty);
 -                } else {
 -                    session.removeAttribute("qty_" + selectedBookId);
 -                    items = items.replace(selectedBookId + ",", "");
 -                    items = items.replace("," + selectedBookId, "");
 -                    items = items.replace(selectedBookId, "");
 -                    session.setAttribute("items", items);
 -                }
 -                else  {
 -                    session.removeAttribute("qty_" + selectedBookId);
 -                    session.setAttribute("items", items.replace(selectedBookId, ""));
 -                }
 +        if (selectedBookId != null) {
 +            if (req.getParameter("addToCart") != null) {
 +                addItemToCart(selectedBookId, session);
 +            } else {
 +                removeItemFromCart(selectedBookId, session);
              }
          }
      }
 +    
 +    private static void addItemToCart(String selectedBookId, HttpSession session) {
 +        String items = (String) session.getAttribute("items");
 +        items = updateItemsList(items, selectedBookId);
 +        session.setAttribute("items", items);
 +        updateItemQuantity(selectedBookId, session, true);
 +    }
 +    
 +    private static void removeItemFromCart(String selectedBookId, HttpSession session) {
 +        int itemQty = getItemQuantity(selectedBookId, session);
 +        if (itemQty > 1) {
 +            updateItemQuantity(selectedBookId, session, false);
 +        } else {
 +            removeItemCompletely(selectedBookId, session);
 +        }
 +    }
 +    
 +    private static String updateItemsList(String items, String selectedBookId) {
 +        if (items == null || items.isEmpty()) {
 +            return selectedBookId;
 +        } else if (!items.contains(selectedBookId)) {
 +            return items + "," + selectedBookId;
 +        }
 +        return items;
 +    }
 +    
 +    private static void updateItemQuantity(String selectedBookId, HttpSession session, boolean increase) {
 +        int itemQty = getItemQuantity(selectedBookId, session);
 +        itemQty = increase ? itemQty + 1 : itemQty - 1;
 +        session.setAttribute("qty_" + selectedBookId, itemQty);
 +    }
 +    
 +    private static int getItemQuantity(String selectedBookId, HttpSession session) {
 +        Object qtyObj = session.getAttribute("qty_" + selectedBookId);
 +        return qtyObj != null ? (int) qtyObj : 0;
 +    }
 +    
 +    private static void removeItemCompletely(String selectedBookId, HttpSession session) {
 +        session.removeAttribute("qty_" + selectedBookId);
 +        String items = (String) session.getAttribute("items");
 +        items = items.replace(selectedBookId + ",", "").replace("," + selectedBookId, "").replace(selectedBookId, "");
 +        session.setAttribute("items", items);
 +    }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged